liota/lib/transports/ENIP.py

```diff
@@ -0,0 +1,103 @@
+#                                                                             #
+#  - Redistributions of source code must retain the above copyright notice,   #
+#      this list of conditions and the following disclaimer.                  #
+#                                                                             #
+#  - Redistributions in binary form must reproduce the above copyright        #
+#      notice, this list of conditions and the following disclaimer in the    #
+#      documentation and/or other materials provided with the distribution.   #
+#                                                                             #
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"#
+#  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE  #
+#  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE #
+#  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE  #
+#  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR        #
+#  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF       #
+#  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS   #
+#  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN    #
+#  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)    #
+#  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF     #
+#  THE POSSIBILITY OF SUCH DAMAGE.                                            #
+# ----------------------------------------------------------------------------#
+
+import logging
+import os
+import sys
+import time
+import cpppo
+from random import randint
+from cpppo.server.enip import client
+from cpppo.server.enip.getattr import attribute_operations
+
+log = logging.getLogger(__name__)
+
+class EtherNetIP:
+	'''
+		EtherNetIP implementation for LIOTA. It uses python-cpppo internally.
+	'''
+	def __init__(self, host=None, port=None, timeout=None, dialect=None, profiler=None, udp=False, broadcast=False, source_address=None):
+		
+		self.host = host
+		self.port = port
+		self.timeout = timeout
+		self.dialect = dialect
+		self.profiler = profiler
+		self.udp = udp
+		self.broadcast =broadcast
+		self.source_address = source_address                                                                               
+		
+
+	def connect(self):
+		with client.connector(host=self.host) as self.conn:
+			if(self.conn):
+				print("Connected to the server")
+		 	log.info("Connected to Server")
+
+
+	def write(self,tag,elements,data,tag_type):
+		self.tag = tag
+		self.elements = elements
+		self.data = data
+		self.tag_type = tag_type
+
+		try:
+			req = self.conn.write( self.tag, elements=self.elements, data=self.data,
+                              tag_type=self.tag_type)
+		except AssertionError:
+            		print "Response timed out!! Tearing Connection and Reconnecting!!!!!"

```

Comment:
Remove all print statement. The error either has to be thrown, or logged. Why are we not throwing the error?

Reply:
I have logged the errors.